src/xmlstr2xmlnode.js

```diff
@@ -24,6 +24,8 @@ const defaultOptions = {
   attributeNamePrefix: '@_',
   attrNodeName: false,
   textNodeName: '#text',
+  appendIndexInNodes: false,
+  indexInNodesName: '#index',
```

Comment:
I would change this to just `index` or `indexXmlPosition`, since you cannot use this prop in a javascript file because props cannot start with `#`, it's a sintaxis error.